### PR TITLE
Fix timezones in ledger xml parsing

### PR DESCRIPTION
**Pull request by dannyroberts**
- Pull ledger xml blocks straight from the form XML, so that they come in the right order
- Apply the same transformations to each block that are applied to form JSON
  - xml2json
  - adjust_datetimes
  - default string conversions
- Separate parsing the json into distinct steps for clarity:
  - figure out which format is being used (individual or per-entry) and pull out the ledger instructions in a consistent format (see `LedgerInstruction` namedtuple)
  - figure out which kind of report type (balance vs. transfer) it is and how to generate `StockTransactionHelper`s given a `LedgerInstruction`.
  - loop through the ledger instructions and yield stock transaction helpers based on the previously selected generator function

### CODE REVIEW

**Comment by dannyroberts** [comment]
@czue @TylerSheffels

File: corehq/apps/commtrack/parsing.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+     "@date": "",
+     "entry": [{"@id": "", "value": [{"@section-id: "", "@quantity": ""},
+               ...]}]}
+
+Conceptually, both formats produce a list of transactions:
+
+    (entity_id="", date="", section_id="", entry_id="", quantity="")
+    ...
+
+but Per-Entry lets you have many different section_ids among the transactions.
```
Comment (czue) [inline corehq/apps/commtrack/parsing.py]: nice doc string!

**Comment by czue** [comment]
nice compromise and clean up!

File: corehq/apps/commtrack/parsing.py (hunk 1 above)
Comment (snopoke) [inline corehq/apps/commtrack/parsing.py]: :+1:

File: corehq/apps/commtrack/parsing.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+                    yield e
+
+    for elem in commtrack_nodes(form_xml):
+        report_type, ledger_json = convert_xml_to_json(
+            elem, last_xmlns=COMMTRACK_REPORT_XMLNS)
+
+        # apply the same datetime & string conversions
+        # that would be applied to XFormInstance.form
+        adjust_datetimes(ledger_json)
+        ledger_json = XFormInstance({'form': ledger_json}).form
```
Comment (snopoke) [inline corehq/apps/commtrack/parsing.py]: Neat trick

Comment (dannyroberts) [inline corehq/apps/commtrack/parsing.py]: Yeah, it initially caught the attention of my "too clever" sensor, but I decided what it's saying exactly semantically (and not just functionally) correct: "Please wrap this field exactly as `XFormInstance` would wrap its `form` field".

Comment (czue) [inline corehq/apps/commtrack/parsing.py]: @snopoke is there a `form_processor` equivalent of doing this call?

Comment (snopoke) [inline corehq/apps/commtrack/parsing.py]: no, though I've been wondering if there should be. Have considered doing something like this:
```python
class FormData(Document):
    form = DictProperty()

FormData({'form': form_data}).form
```

I did once have that here: https://github.com/dimagi/commcare-hq/blob/master/corehq/form_processor/models.py#L194 but it didn't seem to make any difference so I took it out. Could be that we don't have any tests that check the difference.

**Pull request merged**